### DISCUSSION TITLE
How to configure host/domain names via dns?

### DISCUSSION THREAD

**Discussion by a1994sc** [siderolabs/talos#7764] [Q&A] [answered]
Hello, recently started to play with `Talos`, `v1.5.2`, and it has been super cool!

My only problem is that I have not found out how to configure hostname discovery for the nodes....
I have the following dns records in my powerdns dns:

```text
titania-kc1.adrp.xyz.    300    IN      A       10.3.20.10
10.20.3.10.in-addr.arpa. 300    IN      PTR     titania-kc1.adrp.xyz.
```

However talos will always do a random name, per the `machine.features.stableHostname: true` or `talos-<ip>` if `machine.features.stableHostname: false`... Well I know it doesn't matter too much I would prefer to identify the nodes via the names in the records...

I know that I can set it via the matchbox args: `"talos.hostname=titania-kc1"` however that does not persist after it installs it self on the disk....

I could also edit it after it has joined the cluster... but I want to avoid too much manual intervention because I want to automate as much as I can...

I am fine if this is a feature on the roadmap, but I wanted to make sure I was not missing something obvious. Thanks

**Comment by smira** [comment, marked as answer]
Talos doesn't do reverse DNS lookups, the easiest way is to use DHCP server to issue hostnames to the machines. If that is not available, you can put it to the machine configuration via `machibe.network.hostname`.

**Reply by a1994sc** [reply]
Ah... my only problem is that I would need a machine config per node, where right now I just have a control plane and worker machine config... thanks

**Reply by smira** [reply]
You can also patch base machine config on the fly, `talosctl machineconfig patch` is the CLI equivalent of it. This way you keep a base config and customize it per-node.
